src/test/java/de/qabel/desktop/ui/actionlog/ActionlogControllerTest.java

```diff
@@ -1,25 +1,117 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
 
 
 public class ActionlogControllerTest extends AbstractControllerTest {
 
 	ActionlogController controller;
+	Identity i;
+	ActionlogView view;
+	Contact c;
+	String fakeURL;
+	String workingURL;
+	String text = "MessageString";
 
-	@Test
-	public void injectTest() {
-		ActionlogView view = new ActionlogView();
-		Identity i = new Identity("test", null, new QblECKeyPair());
+	@Before
+	public void setUp() throws Exception {
+		super.setUp();
+		fakeURL = "http://localhost:12345/abcdefghijklmnopqrstuvwxyzabcdefgworkingUrl";
+		workingURL = "https://qdrop.prae.me/abcdefghijklmnopqrstuvwxyzabcdefgworkingUrl";
+	}
+
+	private void createController(Identity i, Contact c) {
+
+		view = new ActionlogView();
 		clientConfiguration.selectIdentity(i);
 		clientConfiguration.setAccount(new Account("Provider", "user", "auth"));
 		controller = (ActionlogController) view.getPresenter();
-		assertNotNull(controller);
 	}
+
+	@Test
+	public void sendDropMessageTest() throws QblDropPayloadSizeException {
+		i = identityBuilderFactory.factory().withAlias("TestAlias").build();
+		c = new Contact(i, i.getAlias(), i.getDropUrls(), i.getEcPublicKey());
+		createController(i, c);
+
+		DropMessage dm = controller.sendDropMessage(c, text);
+		assertEquals(text, dm.getDropPayload());
+	}
+
+	@Test
+	public void sendDropMessageFailTest() throws QblDropPayloadSizeException, URISyntaxException, QblDropInvalidURL {
+		Collection<DropURL> collection = new ArrayList<>();
+		DropURL drpoUrl = new DropURL(fakeURL);
+		collection.add(drpoUrl);
+		i = identityBuilderFactory.factory().withAlias("TestAlias").build();
+		c = new Contact(i, i.getAlias(), collection, i.getEcPublicKey());
+		createController(i, c);
+
+		DropMessage dm = controller.sendDropMessage(c, text);
+		assertNull(dm);
+	}
+
+	@Test
+	public void receiveMessagesTest() throws QblDropPayloadSizeException, QblDropInvalidMessageSizeException, QblVersionMismatchException, QblSpoofedSenderException, URISyntaxException, QblDropInvalidURL {
+		String text = "MessageString";
+		Date sinceDate = new Date(0L);
+		Collection<DropURL> collection = new ArrayList<>();
+		DropURL drpoUrl = new DropURL(workingURL);
+		collection.add(drpoUrl);
+
+		i = identityBuilderFactory.factory().withAlias("TestAlias").build();
+		Identity identity = new Identity("TestAlias", collection, i.getPrimaryKeyPair());
+		c = new Contact(identity, identity.getAlias(), collection, identity.getEcPublicKey());
+
+		createController(identity, c);
+		clientConfiguration.selectIdentity(identity);
+		controller = (ActionlogController) view.getPresenter();
+		controller.sendDropMessage(c, text);
+		List<DropMessage> messages = controller.getDropMassages(sinceDate);
+
+		assertEquals(1, messages.size());
+		assertEquals(text, messages.get(0).getDropPayload());
+	}
+
+	@Test
+	public void addMessageToActionlogTest() throws QblDropPayloadSizeException {
```

Comment:
this test doesn't work on my system because no Contact is found which results in a `return null;` that results in strange things within the `injector` and results in a message like
```
java.lang.IllegalStateException: Cannot load de.qabel.desktop.ui.actionlog.item.otheractionlogitem
```

Probably better to debug when the `return null;`s are replaced.